go/oasis-test-runner/scenario/e2e/prune.go

```diff
@@ -0,0 +1,87 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/oasisprotocol/oasis-core/go/oasis-test-runner/env"
+	"github.com/oasisprotocol/oasis-core/go/oasis-test-runner/log"
+	"github.com/oasisprotocol/oasis-core/go/oasis-test-runner/oasis"
+	"github.com/oasisprotocol/oasis-core/go/oasis-test-runner/scenario"
+)
+
+const (
+	pruneNumKept = 20
+)
+
+// Prune is scenario that tests consensus block pruning.
+var Prune scenario.Scenario = &prune{
+	Scenario: *NewScenario("prune"),
+}
+
+type prune struct {
+	Scenario
+}
+
+func (sc *prune) Fixture() (*oasis.NetworkFixture, error) {
+	f, err := sc.Scenario.Fixture()
+	if err != nil {
+		return nil, err
+	}
+
+	f.Validators[0].Consensus.PruneNumKept = pruneNumKept
+	f.Validators[0].Consensus.PruneInterval = time.Second
+	f.Validators[0].LogWatcherHandlerFactories = []log.WatcherHandlerFactory{
+		// Ensure ABCI pruning happens on the node.
+		oasis.LogEventABCIPruneDelete(),
+	}
+
+	return f, nil
+}
+
+func (sc *prune) Clone() scenario.Scenario {
+	return &prune{
+		Scenario: *sc.Scenario.Clone().(*Scenario),
+	}
+}
+
+func (sc *prune) Run(ctx context.Context, _ *env.Env) error {
+	if err := sc.Net.Start(); err != nil {
+		return fmt.Errorf("failed to start network: %w", err)
+	}
+
+	// Wait first few blocks and ensure no pruning.
+	halfPrune := pruneNumKept / 2
+	if _, err := sc.WaitBlocks(ctx, halfPrune); err != nil {
+		return fmt.Errorf("failed waiting %d blocks: %w", halfPrune, err)
+	}
+
+	sc.Logger.Info("ensure no pruning",
+		"start_height", 1,
+		"end_height", halfPrune,
+	)
+	ctrl := sc.Net.Controller()
+	for h := 1; h <= halfPrune; h++ {
+		if _, err := ctrl.Consensus.GetBlock(ctx, int64(h)); err != nil {
+			return fmt.Errorf("pruning too early: %w", err)
+		}
+	}
+
+	// Wait for additional blocks and verify pruning.
+	if _, err := sc.WaitBlocks(ctx, 2*pruneNumKept); err != nil {
+		return fmt.Errorf("failed waiting for %d blocks: %w", 2*pruneNumKept, err)
+	}
+
+	sc.Logger.Info("ensure pruned blocks",
+		"start_height", 1,
+		"end_height", pruneNumKept,
+	)
+	for h := 1; h <= pruneNumKept; h++ {
+		if _, err := ctrl.Consensus.GetBlock(ctx, int64(h)); err == nil { // if NO error
+			return fmt.Errorf("pruning failed: unexpected block (height: %d)", h)
```

Comment:
This error is not true, as you don't now if pruning failed. Maybe it just wasn't executed.